sdk/storage/azure-storage-blob/azure/storage/blob/_models.py

```diff
@@ -836,11 +836,81 @@ class PageRange(DictMixin):
         Start of page range in bytes.
     :param int end:
         End of page range in bytes.
+    :ivar bool cleared:
+        Whether the range has been cleared.
     """
 
-    def __init__(self, start=None, end=None):
+    def __init__(self, start=None, end=None, *, cleared=False):
         self.start = start
         self.end = end
+        self.cleared = cleared
+
+    @staticmethod
+    def _parse_page_list(page_list):
```

Comment:
This should not be a staticmethod - let's move it into a utility function that can be shared between the sync and async iterators.

Reply:
So, I actually initially tried to move this to another helper file (like `_deserialize` but the tricky part is this function depends on `PageRange` so moving it to any other file creates a circular dependency :)

Any suggestions to get around this? Could also just move this to a function at the top of the file.

Reply:
Just at the top of the file is fine :)

Reply:
Sounds good. I've moved it there.